### DISCUSSION TITLE
Using TRACER with FeynArts and FeynCalc

### DISCUSSION THREAD

**Discussion by ammohseni** [FeynCalc/feyncalc#71] [Q&A]
Hello,

I just found this new forum, I am doing two-loop self-energy calculations in a SuperSymmetric model, I have implemented the model in SARAH, generated a FeynArts model using SARAH and wish to use TRACER for reduction of two-loop integrals and finally evaluate them numerically using TSIL.

I wanted to ask if TRACER can be automatically used to reduce amplitudes generated by FeynArts and if so, how can I go about doing it? I could not find any manual for it except the original paper.  Also I will be thankful if you help me with whether TRACER is a good choice and if there are any better options to do this job. Thank you in advance.

Best regards,

**Comment by vsht** [comment]
Hi,

TRACER is a program for reducing 2-loop propagator-type integrals so it
cannot be applied directly to an amplitude. If you somehow evaluate the amplitude
by yourself and extract a list of loop integrals in the TRACER-notation
then you can pass them to TRACER and get some results.

There is no other manual to TRACER than the original paper. It describes exactly
what TRACER does: it takes a set of loop integrals and reduces them to master
integrals.

To my knowledge, there are no publicly available automatic tools for 2-loop calculations in the style
of MadGraph or FormCalc where you enter several commands and get a final result.

We have an example of using FeynCalc to evaluate FeynArts amplitudes at 2-loops

https://feyncalc.github.io/FeynCalcExamplesMD/QCD/TwoLoops/Gh-Gh

but that calculation is admittedly a very simple one (and still it is not entirely
straightforward). 

From my experience, starting with 2-loops you need to employ
FORM for the amplitude evaluation. Mathematica is simply too slow for that
and you will waste infinitely more time trying to optimize a sluggish Mma script
rather than using FORM from the very beginning. FeynCalc is stil useful for some
auxiliary tasks (like deriving tensor reduction rules to be applied by FORM or 
cross-checking single amplitudes), but not as a tool for the full evaluation.

Cheers,
Vladyslav
